src/lib.rs

```diff
@@ -187,6 +202,71 @@ mod tests {
         }
     }
 
+    #[test]
+    fn multi_source_routing_on_random_graph() {
+        const REPEATS: usize = 100;
+        for _ in 0..REPEATS {
+            const NUM_NODES: usize = 50;
+            const NUM_QUERIES: usize = 1_000;
+            const MEAN_DEGREE: f32 = 2.0;
+            const NUM_SOURCES: usize = 3;
+
+            let mut rng = create_rng();
+            let input_graph = InputGraph::random(&mut rng, NUM_NODES, MEAN_DEGREE);
+            debug!("random graph: \n {:?}", input_graph);
+            let fast_graph = prepare(&input_graph);
+            let mut path_calculator = create_calculator(&fast_graph);
+
+            let dijkstra_graph = PreparationGraph::from_input_graph(&input_graph);
+            let mut dijkstra = Dijkstra::new(input_graph.get_num_nodes());
+
+            let mut num_different_paths = 0;
+            for _ in 0..NUM_QUERIES {
+                // This may pick duplicate source nodes; that shouldn't break anything.
+                let sources: Vec<(NodeId, Weight)> = (0..NUM_SOURCES)
+                    .map(|_| (rng.gen_range(0, input_graph.get_num_nodes()), 0))
+                    .collect();
+                let target = rng.gen_range(0, input_graph.get_num_nodes());
+                let path_fast = path_calculator
+                    .calc_path_multiple_endpoints(&fast_graph, sources.clone(), target)
+                    .unwrap_or(ShortestPath::none(sources[0].0, target));
+                let dijkstra_paths: Vec<ShortestPath> = sources
+                    .iter()
+                    .flat_map(|(source, _)| dijkstra.calc_path(&dijkstra_graph, *source, target))
+                    .collect();
+
+                // It's likely that there are multiple shortest paths, from different source nodes.
+                // Compare every Dijkstra's path with the one CH path.
```

Comment:
We have to make sure that `path_fast` is as short as the shortest among `dijkstra_paths`. And yes there can be multiple elements in `dijkstra_paths` with the same (shortest) weight. In this case we have to make sure that `fast_path` equals one of them. Is this what you meant here?